### PR TITLE
RDCC-930, Master: Re-write Exception Tests

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-930

### Change description ###

THESE SAME CHANGES HAVE ALREADY BEEN MERGED TO DEMO: https://github.com/hmcts/rd-user-profile-api/pull/337

Some Unit Tests in User Profile test for Exceptions. Currently, they are throwing the exception before we make a call to the code we are testing. We should be hitting the exception naturally by providing correct mocks to result in an Exception.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[✘] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java [hunk 1]
```diff
@@ -117,72 +107,76 @@ public void testValidateUpdateUserProfileRequestValidHappyPath() {
...
-        sut.validateUserStatusBeforeUpdate(updateUserProfileData, userProfile, API);
+        final Throwable raisedException = catchThrowable(() -> sut.validateUserStatusBeforeUpdate(updateUserProfileData, userProfile, API));
+
+        assertThat(raisedException.getMessage()).contains("User is PENDING or input status is PENDING and only be changed to ACTIVE or SUSPENDED for userId: null");
+        assertThat(raisedException).isInstanceOf(RequiredFieldMissingException.class);
 
         verify(auditServiceMock, times(1)).persistAudit(eq(HttpStatus.BAD_REQUEST), eq(API));
     }
 
     @Test(expected = Test.None.class)
```
Comment (shreedhar-hmcts) [inline src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java:135]: expected should be removed

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java [hunk 2]
```diff
@@ -117,72 +107,76 @@ public void testValidateUpdateUserProfileRequestValidHappyPath() {
...
-    public void testvalidateUserPersistedWithException_scenario1() {
+    public void test_validateUserPersistedWithException_scenario1() {
         assertThat(sut.validateUserPersisted(HttpStatus.OK)).isTrue();
     }
 
-    @Test(expected = Test.None.class)
-    public void testvalidateUserPersistedWithException_scenario2() {
+    @Test(expected = ErrorPersistingException.class)
```
Comment (shreedhar-hmcts) [inline src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java:178]: wondering just changed expected exception here , how original test was passing ?

Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java:178]: May have been that it was not actually testing the method but returning true regardless, if you see the above test (line 174) that is the correct scenario where HttpStatus.OK should return true and not throw exception, but HttpStatus.BAD_REQUEST should throw ErrorPersistingException as it is a Bad Request

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java (hunk 1 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java]: This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/333/commits/3e3d5a6b05b395c1e8271f85b7e7c1f485182407

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java [hunk 3]
```diff
@@ -227,17 +218,16 @@ public void test_validateUserStatusWithException_should_return_true() {
...
         final Throwable raisedException = catchThrowable(() -> sut.validateUserStatus(userProfile, IdamStatus.PENDING));
         assertThat(raisedException).isInstanceOf(InvalidRequest.class);
 
         verify(auditServiceMock, times(1)).persistAudit(any(HttpStatus.class), any(ResponseSource.class));
         verify(exceptionServiceMock, times(1)).throwCustomRuntimeException(any(ExceptionType.class), any(String.class));
     }
 
-    @Test(expected = Test.None.class)
+    @Test()
     public void test_validateUserLastUpdatedWithinSpecifiedTimeWithException_should_return_true() {
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java:231]: did  we required brackets here?

Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java:231]: No, will remove

Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java:231]: This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/333/commits/4f75270dfe37cfc602014c621a2b4443053a63f5

**Pull request merged**